### PR TITLE
notifications: fix request and recall

### PR DESCRIPTION
**Pull request by jma**
* Does not create a request notification when the item is on loan.
* Prevents sending multiple recall notifications.
* Fixes bad notification emails in the tests fixtures.
* Ensures booking notification are always sent.
* Fixes email recipient for transit notice and booking notifications.
* Improves availiblity notification templates.

_Note:_ Fixes A, B, D of #2152.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by benerken — APPROVED** [review]

**Comment by iGormilhit** [comment]
Fixes A, B, D of rero/rero-ils#2152.

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/loans/api.py [hunk 1]
```diff
@@ -761,10 +761,12 @@ def create_notification(self, notification_type=None, counter=0):
...
 
-        # recall + availibility
+        # availibility
         if notif_type in [
-                Notification.RECALL_NOTIFICATION_TYPE,
-                Notification.AVAILABILITY_NOTIFICATION_TYPE]:
+                Notification.AVAILABILITY_NOTIFICATION_TYPE,
+                # recall
+                Notification.RECALL_NOTIFICATION_TYPE
+        ]:
```
Comment (zannkukai) [inline rero_ils/modules/loans/api.py:769]: COSMETIC : Is it necessary to change ?

File: rero_ils/modules/notifications/api.py [hunk 2]
```diff
@@ -207,6 +202,19 @@ def init_loan(self):
             self.loan = Loan.get_record_by_pid(self.loan_pid)
         return self.loan
 
+    def _get_library_informations(self, library_pid):
+        """."""
+        library = Library.get_record_by_pid(library_pid)
+        data = dict()
+        for field in [
+            'pid', 'email', 'name', 'address', 'notification_settings',
+            'communication_language', 'next_open'
```
Comment (zannkukai) [inline rero_ils/modules/notifications/api.py:211]: `next_open` isn't a field, it's a function. So you couldn't use `library.get('next_open')`.
I think you need to keep original code 
```
keep_until = datetime.now(timezone.utc) + timedelta(days=10)
next_open = library.next_open(keep_until)
```

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: rero_ils/modules/notifications/api.py (hunk 2 above)
Comment (iGormilhit) [reply rero_ils/modules/notifications/api.py:211]: Thanks for reviewing!

**Review by zannkukai — APPROVED** [review]

**Pull request merged**